wagtail/contrib/settings/templatetags/wagtailsettings_tags.py

```diff
@@ -12,10 +12,14 @@
 def get_settings(context, use_default_site=False):
     if use_default_site:
         site = Site.objects.get(is_default_site=True)
-        context['settings'] = SettingsProxy(site)
+        settings = SettingsProxy(site)
     elif 'request' in context:
-        context['settings'] = SettingsProxy(context['request'])
+        settings = SettingsProxy(context['request'])
     else:
         raise RuntimeError('No request found in context, and use_default_site '
                            'flag not set')
-    return ''
+    # For backward compatibility with the old behaviour of this tag
+    # set context['settings'] if it isn't already set
+    if 'settings' not in context:
+        context['settings'] = settings
```

Comment:
Python has a built-in mechanism for inserting into a dict only when the key doesn't exist

```suggestion
    context.setdefault('settings', settings)
```

Reply:
Thanks!